tools/git.py

```diff
@@ -74,6 +78,11 @@ def run(self):
             args = ["git", "checkout", self.commit]
             logging.info("running command: %s" % " ".join(args))
             sp.run(args, cwd=repository_dir, check=True)
+        if self.patches is not None:
```

Comment:
It is not clear to me how this would work via git apply, is this tested with an example?

Reply:
You mean that the patch is applied properly, right? Yes, I did test this.

Reply:
See documentation of `git apply` [here](https://git-scm.com/docs/git-apply):
>  \- can be used to read from the standard input.

Therefore I pass the patch as a string as input.